src/service-implementations/lf-repo-tree-node-service.ts

```diff
@@ -272,33 +276,38 @@ export class LfRepoTreeNodeService implements LfTreeNodeService {
    * };
    *  ```
    */
-  createLfRepoTreeNode(entry: Entry, repoName: string, parent?: LfRepoTreeNode): LfRepoTreeNode {
+  createLfRepoTreeNodeWithParent(entry: Entry, parent: LfRepoTreeNode): LfRepoTreeNode {
     let treeNode: LfRepoTreeNode | undefined;
     if (!entry.entryType) {
       throw new Error('entry type is undefined');
     }
-    if (entry.entryType === EntryType.Folder) {
-      treeNode = this.createFolderNode(entry, repoName, parent);
-    } else if (entry.entryType === EntryType.Document) {
-      treeNode = this.createLeafNode(entry, parent);
-    } else if (entry.entryType === EntryType.Shortcut) {
-      const shortcut = entry as Shortcut;
-      if (shortcut.targetType === EntryType.Folder) {
-        treeNode = this.createFolderNode(shortcut, repoName, parent);
-      } else if (shortcut.targetType === EntryType.Document) {
-        treeNode = this.createLeafNode(shortcut, parent);
-      }
-      else if (shortcut.targetType === EntryType.RecordSeries) {
-        treeNode = this.createFolderNode(shortcut, repoName, parent);
-      }
-      else {
-        throw new Error('Unexpected shortcut targetType');
-      }
-    } else if (entry.entryType === EntryType.RecordSeries) {
-      treeNode = this.createFolderNode(entry, repoName, parent);
-    }
-    else {
-      throw new Error('Unsupported entry type');
+    switch (entry.entryType) {
+      case EntryType.Folder:
+        treeNode = this.createFolderNodeWithParent(entry, parent);
+        break
+      case EntryType.RecordSeries:
+        treeNode = this.createFolderNodeWithParent(entry, parent);
+        break
+      case EntryType.Document:
+        treeNode = this.createLeafNode(entry, parent);
+        break
+      case EntryType.Shortcut:
+          switch ((entry as Shortcut).targetType) {
+            case EntryType.Folder:
```

Comment:
this seems to be working and it was already like this, but I'm missing where targetId is getting assigned on the repoTreeNode.

Reply:
targetId is assigned in setNodeProperties called in this function. We could probably refactored this so it is easier to read.